### ISSUE TITLE
Spock support

### ISSUE THREAD

**Issue by mpodlodowski** [Microsoft/vscode-java-test#425]
Would be great to have Spock support. I am very willing to make a contribution.
Is there anything I should be aware of before digging into the problem?

**Comment by jdneo** [comment]
@mpodlodowski PR is welcome. 😄 

Please feel free to ask questions whenever you meet any problems or need any help

**Comment by RaviH** [comment]
Hi @mpodlodowski,

Any luck?

**Comment by mpodlodowski** [comment]
Just met some problems trying to debug the Java code - I can't attach to remote debugging port 1044. Is the guide on cs/contributing still valid?

**Comment by jdneo** [comment]
@mpodlodowski Are you debugging the code under `com.microsoft.java.test.plugin` or `com.microsoft.java.test.runner`

**Comment by mpodlodowski** [comment]
`com.microsoft.java.test.plugin`

At least I'm trying to. Is it possible?

**Comment by jdneo** [comment]
Yes, it should be possible. Did you launch the extension first then attach to the remote application?

**Comment by mpodlodowski** [comment]
I did (with "Launch Extension" config from the repo). 

I tried the following:
1. gulp build-server
2. Launch Extension
3. attach debugger

Am I missing something?

**Comment by jdneo** [comment]
Did you set the target platform in Eclipse when importing the `com.microsoft.java.test.plugin`?

After importing the project, you can find a file called [target.target](https://github.com/Microsoft/vscode-java-test/blob/master/java-extension/com.microsoft.java.test.plugin/target.target). Double click it in Eclipse, and in the right top of the editor, click `Set as active target platform`

**Comment by mpodlodowski** [comment]
Done. Still no luck.
```
Failed to connect to remote VM. Connection refused.
Connection refused (Connection refused)
```

**Comment by jdneo** [comment]
Hmm, another thing needs to pay attention is that we need to wait until the extension has been activated, then we can attach to the remote VM using Eclipse.

You can check whether the extension has been activated by checking if the `Java Test Runner` output channel has been created.

**Comment by sapelkinAV** [comment]
Any news guys?)

**Comment by napalm23zero** [comment]
Any News Guys????

**Comment by TomerFi** [comment]
This would be very much appreciated...

**Comment by juusu** [comment]
Any progress? Just started on a new project which uses primarily Spock for testing so this would come in very handy

**Comment by jdneo** [comment]
Hi friends, so far there is no plan to add more test framework support. But contribution is welcome!

> Note that VS Code is now working on [a new set of APIs for test purpose](https://github.com/microsoft/vscode/issues/107467), which means the extension might need to have a big refactor in the near future, so I would suggest at least wait until that happens.

**Comment by Szwajcii** [comment]
Hello ;) Any news about Spock framework support? ;)

**Comment by changeling** [comment]
Just bumping the above comment. Any word?

**Comment by tiagohumbs** [comment]
Hello... Whats is news ?

**Comment by SeanCodeMedia** [comment]
Hello is there any new updates on Spock support ?

**Comment by iaincalderfh** [comment]
Would also really appreciate Spock support

**Comment by wyy0117** [comment]
Six years later, it still doesn't support Spock

**Comment by vnguyen972** [comment]
There has to be a smart person (smarter than me) somewhere :) for a Spock Test Runner plugin!

**Comment by pyaqubzade** [comment]
Seven years later, it still doesn't support Spock
